nixos/modules/services/hardware/udev.nix

```
@@ -167,10 +167,17 @@ let
       mv etc/udev/hwdb.bin $out
     '';
 
-  compressFirmware = firmware: if (config.boot.kernelPackages.kernelAtLeast "5.3" && (firmware.compressFirmware or true)) then
-    pkgs.compressFirmwareXz firmware
-  else
-    id firmware;
+  compressFirmware = firmware:
+    let
+      inherit (config.boot.kernelPackages) kernelAtLeast;
+      compress =
+        if ! (firmware.compressFirmware or true) then
+          id
+        else
+          if kernelAtLeast "5.19" then pkgs.compressFirmwareZstd
+          else if kernelAtLeast "5.3" then pkgs.compressFirmwareXz
+          else id;
+    in compress firmware;
```

nit: no need to wrap this all in an extra lambda
```suggestion
  compressFirmware = let
      inherit (config.boot.kernelPackages) kernelAtLeast;
    in
      if ! (firmware.compressFirmware or true) then
        id
      else
        if kernelAtLeast "5.19" then pkgs.compressFirmwareZstd
        else if kernelAtLeast "5.3" then pkgs.compressFirmwareXz
        else id;
```